wasm/src/main/scala/ir2wasm/HelperFunctions.scala

```diff
@@ -524,6 +505,83 @@ object HelperFunctions {
     fctx.buildAndAddToContext()
   }
 
+  /** `newArrayOfThisClass: (ref typeData), anyref -> (ref jlObject)`.
+    *
+    * This is the underlying func for the `newArrayOfThisClass()` closure inside class data objects.
+    */
+  private def genNewArrayOfThisClass()(implicit ctx: WasmContext): Unit = {
```

Comment:
[note]
Takes a component type and array dimensions (in JS array), allocates the necessary metadata to describe the array type, and then calls a method to allocate a new array object of that type and dimensions (`newArrayObject`).